compiler/ir/ir.psi2ir/src/org/jetbrains/kotlin/psi2ir/generators/GeneratorContext.kt

```diff
@@ -31,7 +32,8 @@ class GeneratorContext private constructor(
     val extensions: GeneratorExtensions,
     val typeTranslator: TypeTranslator,
     override val irBuiltIns: IrBuiltIns,
-    internal val callToSubstitutedDescriptorMap: MutableMap<IrDeclarationReference, CallableDescriptor>
+    internal val callToSubstitutedDescriptorMap: MutableMap<IrDeclarationReference, CallableDescriptor>,
+    internal val capturedDescriptorToFragmentParameterMap: MutableMap<DeclarationDescriptor, IrValueParameterSymbol>
```

Comment:
Let's slightly encapsulate this in a new class (`FragmentContext` or something like that?).

Among increasing readability a bit, it'll allow to make the map nullable and mutable, and initialized to null at the beginning, so that we could completely skip hash table lookup in `CallGenerator.generateValueReference` in the non-fragment case.

Reply:
Done. Let me know if my take on this is what you had in mind. 

I don't see the need for the _map_ to be nullable, but I guess it's just a matter of where you put the `?`: on the additional context or the map :)

Reply:
Will let you know when I am done with this first pass, so you can address all this at once.

Reply:
👍